exercises/binary-search-tree/canonical-data.json

```diff
@@ -0,0 +1,68 @@
+{
+    "exercise": "binary-search-tree",
+    "version": "1.0.0",
+    "comments": [
+        "For simplicity only positive integer is considered as valid data,",
+        "null should be replaced by the respective language equivalent or -1",
+        "if the language does not have any.",
+
+        "The purpose of ths exercise is to create a data structure which",
+        "makes sure the data is always sorted, hence the exercise should",
+        "not make use of any sort() functions or sorting algorithms."
+    ],
+    "cases": [{
+            "description": "insert invalid/null data into the tree",
+            "property": "insert",
+            "value": null,
+            "expected": false
+        },
+        {
+            "description": "insert a number into the tree",
+            "property": "insert",
+            "value": 6,
+            "expected": true
+        },
+        {
+            "description": "insert a number that already exist",
+            "property": "insert",
+            "value": 6,
```

Comment:
I didn't see how we can understand that the number already exists, using the information provided in this case. If we suppose that we are inserting into a tree that already had `6` inserted into it at some point, that needs to be specified in some way here. Otherwise, compare this case with the case immediately above it: the only difference is the `description` and `expected` values, so there doesn't seem to be any apparent reason why one would expect true and the other would expect false.